common/Resources/ti.internal/extensions/node/fs.js

```diff
@@ -0,0 +1,1622 @@
+		}
+	}
+	const tiMode = FLAGS_TO_TI_MODE.get(flags);
+	// TODO: If invalid flag, throw an error!
+	return createFileDescriptor(path, tiFile.open(tiMode));
+};
+
+/**
+ * @callback readCallback
+ * @param {Error} err - Error if one occurred
+ * @param {integer} bytesRead - number of bytes read
+ * @param {Buffer} buffer buffer
+ */
+
+/**
+ * @param {integer} fd file descriptor
+ * @param {Buffer|Ti.Buffer} buffer buffer to read into
+ * @param {integer} offset the offset in the buffer to start writing at.
+ * @param {integer} length integer specifying the number of bytes to read.
+ * @param {integer} position where to begin reading from in the file
+ * @param {readCallback} callback async callback
+ */
+fs.read = (fd, buffer, offset, length, position, callback) => {
+	const tiFileStream = streamForDescriptor(fd);
+	if (!Buffer.isBuffer(buffer)) {
+		buffer = Buffer.from(buffer);
+	}
+	tiFileStream.read(buffer.toTiBuffer(), offset, length, readObj => {
+		if (!readObj.success) {
+			callback(new Error(readObj.error));
+			return;
+		}
+		callback(null, readObj.bytesProcessed, buffer);
+	});
+};
+
+/**
+ * @param {integer} fd file descriptor
+ * @param {Buffer|Ti.Buffer} buffer buffer to read into
+ * @param {integer} offset the offset in the buffer to start writing at.
+ * @param {integer} length integer specifying the number of bytes to read.
+ * @param {integer} _position where to begin reading from in the file
+ * @returns {integer} bytes read
+ */
+fs.readSync = (fd, buffer, offset, length, _position) => {
+	// FIXME: Allow using position argument!
+	const fileStream = streamForDescriptor(fd);
+	if (!Buffer.isBuffer(buffer)) {
+		buffer = Buffer.from(buffer);
+	}
+	return fileStream.read(buffer.toTiBuffer(), offset, length);
+};
+
+/**
+ * @callback filesCallback
+ * @param {Error} err - Error if one occurred
+ * @param {string[]|Buffer[]|fs.Dirent[]} files - file listing
+ */
+
+/**
+ * @param {string} path directory to list
+ * @param {string|object} [options] optional options
+ * @param {string} [options.encoding='utf8'] encoding to use for filenames, if `'buffer'`, returns `Buffer` objects
+ * @param {boolean} [options.withFileTypes=false] if true, returns `fs.Dirent` objects
+ * @param {filesCallback} callback async callback
+ */
+fs.readdir = (path, options, callback) => {
+	setTimeout(() => {
+		if (typeof options === 'function') {
+			callback = options;
+			options = {};
+		}
+
+		let result;
+		try {
+			result = fs.readdirSync(path, options);
+		} catch (e) {
+			callback(e);
+			return;
+		}
+		callback(null, result);
+	}, 1);
+};
+
+/**
+ * @param {string} path directory to list
+ * @param {string|object} [options] optional options
+ * @param {string} [options.encoding='utf8'] encoding to use for filenames, if `'buffer'`, returns `Buffer` objects
+ * @param {boolean} [options.withFileTypes=false] if true, returns `fs.Dirent` objects
+ * @returns {string[]|Buffer[]|fs.Dirent[]}
+ */
+fs.readdirSync = (path, options) => {
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/fs.js;
  Line: 696;
-->

- :warning: common/Resources/ti.internal/extensions/node/fs.js line 696 – 'options' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)